pkg/service/controller/ingressip/service_ingressip_controller.go

```diff
@@ -167,30 +166,22 @@ func (ic *IngressIPController) enqueueChange(new interface{}, old interface{}) {
 // Run begins watching and syncing.
 func (ic *IngressIPController) Run(stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
-	go ic.controller.Run(stopCh)
+	defer ic.queue.ShutDown()
 
 	glog.V(5).Infof("Waiting for the initial sync to be completed")
-	for !ic.controller.HasSynced() {
-		select {
-		case <-time.After(SyncProcessedPollPeriod):
-		case <-stopCh:
-			return
-		}
+	if !cache.WaitForCacheSync(stopCh, ic.hasSynced) {
+		return
 	}
 
 	if !ic.processInitialSync() {
 		return
 	}
 
-	glog.V(5).Infof("Starting normal worker")
-	for {
-		if !ic.work() {
-			break
-		}
+	glog.V(5).Infof("Starting worker")
+	for ic.work() {
```

Comment:
finish our normal structure? I can't see at a glance here if this handles panics properly with a backoff and retry.